### PR TITLE
Typescript: Create shared config

### LINKED ISSUE #18008: Packages: create shared tsconfig for Grafana packages
Right now every package has its own tsconfig.json file. We should extract the common part to packages/tsconfig.json and extend that file in the packages.

ref: https://github.com/microsoft/vscode/tree/master/extensions

### PR DESCRIPTION
**Pull request by tskarhed**
Closes #18008

### CODE REVIEW

File: packages/grafana-data/tsconfig.json [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
-    "declaration": true,
-    "declarationDir": "dist",
-    "strict": true,
-    "alwaysStrict": true,
-    "noImplicitAny": true,
-    "strictNullChecks": true,
     "typeRoots": ["./node_modules/@types", "types"],
-    "skipLibCheck": true, // Temp workaround for Duplicate identifier tsc errors,
-    "removeComments": false
+    "declarationDir": "dist",
```
Comment (dprokop) [inline packages/grafana-data/tsconfig.json]: Can't this be shared? I assume it makes the `dist` being relative to the shared tsconfig.json location, righht?

**Review by dprokop — COMMENTED** [review]
There's also grafana-toolkit package that it would be good to use the shared tsconfig too.

File: packages/grafana-data/tsconfig.json (hunk 1 above)
Comment (dehrax) [reply packages/grafana-data/tsconfig.json]: >All relative paths found in the configuration file will be resolved relative to the configuration file they originated in.

So yeah, that's the case. One solution might be to write `**/dist`?

Comment (dprokop) [reply packages/grafana-data/tsconfig.json]: I'm not sure this would work. Ok let's leave it as it is now.

**Review by dprokop — APPROVED** [review]

**Pull request merged**